### PR TITLE
feat(samples): add Discord client sample

### PR DESCRIPTION
**Pull request by pashafateev**
Related to https://github.com/autokitteh/samples/pull/132, I propose providing two samples because Discord is unique in having event-driven capabilities in its client.

### CODE REVIEW

File: discord/discord_client/autokitteh.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+project:
+  name: discord_client
+  vars: 
+    - name: CHANNEL_ID
+      value: "YOUR_CHANNEL_ID" 
+  connections:
+    - name: discord_connection
+      integration: discord
+    - name: http_connection
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/autokitteh.yaml]: Done

File: discord/discord_client/program.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Modules:
+    - autokitteh.discord: Custom wrapper for the Discord API client.
+    - discord: The native Discord client API.
+
+Functions:
+    - on_ready: An asynchronous function that is triggered when the bot successfully connects to Discord.
+        It checks for the availability of a specified channel and sends a message if the channel is found.
+        Handles permission and HTTP-related exceptions when attempting to send the message.
+
+    - on_http_get: An Autokitteh activity that runs the bot when an HTTP GET request is received.
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/program.py]: Done

File: discord/discord_client/program.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            await channel.send("Meow!")
+        except discord.Forbidden:
+            print("The bot does not have permission to send messages in this channel.")
+        except discord.HTTPException as e:
+            print(f"Failed to send message: {e}")
+
+
+@autokitteh.activity
+def on_http_get(event):
+    client.run(ak_discord.get_bot_token(CONNECTION_NAME))
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/program.py]: Done

File: discord/discord_client/autokitteh.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# This YAML file is a declarative manifest that describes the setup
+# of an AutoKitteh sample project that demonstrates integration with
+# Discord (https://discord.com/).
+#
+# Before deploying this AutoKitteh project, set "CHANNEL_ID"
+# to your own Discord Channel.
+# 
+# After applying this file, initialize this AutoKitteh project's
+# Discord connection.
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/autokitteh.yaml]: Done

File: discord/discord_client/autokitteh.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+version: v1
+
+project:
+  name: discord_client
+  vars: 
+    - name: CHANNEL_ID
+      value: "YOUR_CHANNEL_ID"
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/autokitteh.yaml]: Yup. Forgot about that. Done

File: discord/discord_client/autokitteh.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+version: v1
+
+project:
+  name: discord_client
+  vars: 
+    - name: CHANNEL_ID
+      value: "YOUR_CHANNEL_ID" 
+  connections:
+    - name: discord_connection
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/autokitteh.yaml]: Done

File: discord/discord_client/autokitteh.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      integration: discord
+    - name: http_connection
+      integration: http
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/autokitteh.yaml]: Done

File: discord/discord_client/autokitteh.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  triggers:
+    - name: http_trigger
+      connection: http_connection
+      # Triggered by GET request to http://localhost:9980/http/discord_client
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/autokitteh.yaml]: Done

File: discord/discord_client/program.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+"""
+
+import os
+
+import autokitteh
+import autokitteh.discord as ak_discord
+import discord
+
+CHANNEL_ID = os.getenv("CHANNEL_ID")
+CONNECTION_NAME = "discord_connection"
```
[the remark this answers is not in the document]
Comment (pashafateev) [reply discord/discord_client/program.py]: Done

File: discord/discord_client/program.py [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        for guild in client.guilds:
+            for channel in guild.channels:
+                print(f"{channel.name} (ID: {channel.id})")
+    else:
+        try:
+            await channel.send("Meow!")
+        except discord.Forbidden:
+            print("The bot does not have permission to send messages in this channel.")
+        except discord.HTTPException as e:
+            print(f"Failed to send message: {e}")
```
Comment (daabr) [inline discord/discord_client/program.py:62]: Less indentation = better.

```suggestion
                print(f"{channel.name} (ID: {channel.id})")
        return

    try:
        await channel.send("Meow!")
    except discord.Forbidden:
        print("The bot does not have permission to send messages in this channel.")
    except discord.HTTPException as e:
        print(f"Failed to send message: {e}")
```

**Review by daabr — APPROVED** [review]

**Pull request merged**